addons/delivery/models/delivery_carrier.py

```diff
@@ -137,7 +137,7 @@ def get_price(self):
                 else:
                     computed_price = 0.0
 
-            self.price = computed_price * (1.0 + (float(self.margin) / 100.0))
+            self.price = float(computed_price) * (1.0 + (float(self.margin) / 100.0))
 
```

Comment:
Suggestion: the best fix would be fix from the root, not on the spot.
Try fixing return [here](https://github.com/odoo-dev/odoo/blob/634e2b4e1e5bc8d0643b4f441f99be98e346bead/addons/delivery/models/delivery_carrier.py#L148),  with return [0.0]